core/controllers/suggestion.py

```diff
@@ -83,13 +89,65 @@ class SuggestionHandler(base.BaseHandler):
 
     @acl_decorators.can_suggest_changes
     def post(self):
+        change_dict = self.payload.get('change')
+        target_id = self.payload.get('target_id')
+        target_type = self.payload.get('target_type')
+        target_version_at_submission = self.payload.get(
+            'target_version_at_submission')
+        suggestion_type = self.payload.get('suggestion_type')
+        description = self.payload.get('description')
+
+        suggestion_domain_class = (
+            suggestion_registry.SUGGESTION_TYPES_TO_DOMAIN_CLASSES[
+                suggestion_type])
+        # While creating the below domain object fields suggestion_id and
+        # score_category is kept None because values for these fields are
+        # created in the create_suggestion() method. Also the domain object
+        # created below will only be used for extracting the HTML from the
+        # suggestions and these fields are not required for HTML extraction.
+        suggestion = suggestion_domain_class(
+            None, target_id, target_version_at_submission,
+            suggestion_models.STATUS_IN_REVIEW, self.user_id, None, change_dict,
+            None)
+        html_list = suggestion.get_all_html_content_strings()
+        filenames = (
+            html_cleaner.get_image_filenames_from_html_strings(html_list))
+        # In the case of edit state content suggestion the images are already
+        # saved to GCS and we don't need to save them in this controller.
+        if suggestion_type == (
+                suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT):
+            filenames = []
+        if len(filenames) > 0:
+            suggestion_image_entity_type = (
+                fs_services.get_entity_type_for_suggestion_target(target_type))
+        for filename in filenames:
+            image = self.request.get(filename)
+            if not image:
+                logging.error(
+                    'Image not provided for file with name %s when the '
+                    ' suggestion with target id %s was created.' % (
+                        filename, target_id))
+                raise self.InvalidInputException(
+                    'No image data provided for file with name %s.'
+                    % (filename))
+            try:
+                file_format = (
+                    image_validation_services.validate_image_and_filename(
+                        image, filename))
+            except utils.ValidationError as e:
+                e = '%s' % (e)
```

Comment:
Just inline this below (don't redefine vars).

Reply:
done